File: backend/domain/src/main/java/org/art/vertex/domain/tag/model/Tag.java
package org.art.vertex.domain.tag.model;

import lombok.Builder;
import lombok.Value;

import java.time.LocalDateTime;
import java.util.UUID;

@Value
@Builder
public class Tag {

    UUID id;

    UUID userId;

    String name;

    LocalDateTime createdAt;

    LocalDateTime updatedAt;

    Integer version;

    public static Tag create(UUID id, UUID userId, String name, LocalDateTime ts) {
        return Tag.builder()
            .id(id)
            .userId(userId)
            .name(name)
            .createdAt(ts)
            .updatedAt(ts)
            .version(null)
            .build();
    }
}

File: frontend/packages/types/src/domain/Tag.ts
export interface Tag {
  id: string;
  userId: string;
  name: string;
  description: string | null;
  color: string | null;
  createdAt: string;
  updatedAt: string;
  version: number;
}

export interface CreateTagData {
  name: string;
  description?: string;
  color?: string;
}

export interface UpdateTagData {
  name?: string;
  description?: string;
  color?: string;
}
